### PR TITLE
Upload files to unique folder on server to avoid conflicts

### LINKED ISSUE #275: Upload files with unique names
We use upload_file_in_tmp_folder to upload files to the dpf server. This uploads all the files to the same folder which means files with the same name get overwritten. This is in particular problematic with multiple concurrent clients and assemblies.
We could either upload the files with a unique name or create unique folder names for related files. Folders can be created like this:

tmp_dir = dpf.make_tmp_dir_server() # Returns the dpf tmp folder (only one per server)
tmp_dir_random = os.path.join(tmp_dir, "random_name")
name_on_server1 = dpf.upload_file(rst_path1, os.path.join(tmp_dir_random, "file.rst")

### PR DESCRIPTION
**Pull request by janvonrickenbach**
closes #275 

I decided to just put every file in a separate folder with a unique name. In this way we preserver the descriptive filenames and the unique folder name acts as kind of a prefix.

Tested with a local windows server (inprocess), windows grpc server and linux (docker) server from windows client

### CODE REVIEW

File: src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py [hunk 1]
```diff
@@ -11,6 +13,31 @@
 )
 
 
+def upload_file_to_unique_folder(path_on_client: _PATH, server: BaseServer) -> str:
+    """Upload file to a unique folder on the server.
```
Comment (greschd) [inline src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:17]: Should we make it explicit that this is still a _temporary_ folder? Maybe even include it in the name?

**Review by greschd — APPROVED** [review]

File: tests/element_info_test.py [hunk 2]
```diff
@@ -48,15 +48,20 @@ def test_section_definitions_from_multiple_sources(dpf_server):
         files, server=dpf_server, default_unit_system=unit_systems.solver_nmm
     )
 
+    ud_mat_id = composite_model.material_names["Epoxy Carbon UD (230 GPa) Wet"]
```
Comment (janvonrickenbach) [inline tests/element_info_test.py:51]: We cannot rely on dpf mat ids in tests. Dpf mat ids were different when using a windows grpc server.

File: src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py [hunk 3]
```diff
@@ -11,6 +13,34 @@
 )
 
 
+def upload_file_to_unique_tmp_folder(path_on_client: _PATH, server: BaseServer) -> str:
+    """Upload file to a unique temporary folder on the server.
+
+    Parameters
+    ----------
+    path_on_client:
+        file path on the client side to upload
```
Comment (PipKat) [inline src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:22]:
```suggestion
        Ffile path on the client side to upload the file to.
```

File: src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py [hunk 4]
```diff
@@ -11,6 +13,34 @@
...
+    server:
+        Dpf server
```
Comment (PipKat) [inline src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:24]:
```suggestion
        DPF server.
```

File: src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py [hunk 5]
```diff
@@ -11,6 +13,34 @@
...
+    if server.os == "posix":
+        path_on_server = str(tmp_dir) + "/" + str(uuid.uuid4()) + "/" + path_on_client.name
+    else:
+        path_on_server = str(tmp_dir) + "\\" + str(uuid.uuid4()) + "\\" + path_on_client.name
+
+    uploaded_path = cast(str, dpf.upload_file(path_on_client, path_on_server, server=server))
+    if uploaded_path == "":
+        raise RuntimeError(
+            f"Failed to upload file {path_on_client} to server. "
+            f"Attempted to upload to {path_on_server}"
```
Comment (PipKat) [inline src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:39]: Messages should conclude with punctuation.

Comment (janvonrickenbach) [inline src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:39]:
```suggestion
            f"Attempted to upload to {path_on_server}."
```

File: src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py [hunk 6]
```diff
@@ -11,6 +13,34 @@
...
+        Client side path of the file which should be uploaded to the server.
+    server:
+        DPF server.
+    """
+    tmp_dir = pathlib.Path(
+        dpf.make_tmp_dir_server()
+    )  # Returns the dpf tmp folder (only one per server)
+    path_on_client = pathlib.Path(path_on_client)
+    if server.os == "posix":
+        path_on_server = str(tmp_dir) + "/" + str(uuid.uuid4()) + "/" + path_on_client.name
```
[the remark this answers is not in the document]
Comment (janvonrickenbach) [reply src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:31]: path_on_client is the full path. Do you mean `path_on_client.name / uuid.uuid4()`? In this case the paths would look a bit strange (with file extentensions):

/tmp/dataProcessing21351/file.rst/1234-1512-1512

It's a bit unconventional but I don't see why it should not work. I will give it a try.

[the remark this answers is not in the document]
Comment (janvonrickenbach) [reply src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:31]: Dpf is using the file extension to load the correct operator (e.g. for rst files) and fails if we just add a suffix. I think I'll leave it as it is.

[the remark this answers is not in the document]
Comment (roosre) [reply src/ansys/dpf/composites/server_helpers/_upload_files_to_server.py:31]: `/tmp/dataProcessing21351/file.rst/1234-1512-1512` does not make sense for me. I thought about /tmp/<my local path>/<uuid>/<file name>. For instance `/tmp/dataProcessing21351/1234-1512-1512//file.rst`. But as I said. It's a cosmetical thing.

**Pull request merged**